src/MongrationDotNet/MigrationRunner.cs

```diff
@@ -9,47 +10,58 @@ public class MigrationRunner : IMigrationRunner
     {
         private readonly IMongoDatabase database;
         private readonly ILogger<MigrationRunner> logger;
-        private readonly IEnumerable<IMigration> migrationCollections;
+        private readonly IEnumerable<IMigration> migrationCollection;
         private IMongoCollection<MigrationDetails> migrationDetailsCollection;
 
-        public MigrationRunner(IMongoDatabase database, IEnumerable<IMigration> migrationCollections,
+        public MigrationRunner(IMongoDatabase database, IEnumerable<IMigration> migrationCollection,
             ILogger<MigrationRunner> logger = null)
         {
             this.database = database;
-            this.migrationCollections = migrationCollections;
+            this.migrationCollection = migrationCollection;
             this.logger = logger;
         }
 
         public async Task Migrate()
         {
             migrationDetailsCollection = database.GetCollection<MigrationDetails>(Constants.MigrationDetailsCollection);
 
-            foreach (var migrationCollection in migrationCollections)
+            foreach (var migration in migrationCollection)
             {
                 logger?.LogInformation(LoggingEvents.MigrationStarted,
                     "Migration started for type: {type}, version: {version} and description: {description} ",
-                    migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
+                    migration.Type, migration.Version.ToString(), migration.Description);
                 
                 var latestAppliedMigration = await migrationDetailsCollection
-                    .Find(x => x.Version == migrationCollection.Version)
+                    .Find(x => x.Version == migration.Version)
```

Comment:
If the first time i start my app and it runs migration, a migration errors then the migration is abandond. However if i restart the app, migrations will run again and then this time the errored migration 1 is skipped. Is that what you want? Shouldn't it retry the migration?

Reply:
Added option to retry migration